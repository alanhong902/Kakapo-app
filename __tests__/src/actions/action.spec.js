import configureStore from 'redux-mock-store';
import {
  searchActions,
  settingActions,
  soundActions,
  themeActions,
  notifyActions,
} from 'actions/';

test('[action/searchActions]', () => {
  const store = configureStore([])({});
  store.dispatch(searchActions.input('test'));
  store.dispatch(searchActions.kakapo());
  store.dispatch(searchActions.youtube('test'));
  store.dispatch(searchActions.soundcloud('test'));
  store.dispatch(searchActions.request());
  store.dispatch(searchActions.requestSuccess('test', 'test'));
  store.dispatch(searchActions.requestError('test'));
  expect(store.getActions()).toMatchSnapshot();
});

test('[action/settingActions]', () => {
  const store = configureStore([])({});
  store.dispatch(settingActions.mute('test'));
  store.dispatch(settingActions.dock('test'));
  store.dispatch(settingActions.devtools('test'));
  store.dispatch(settingActions.language('test'));
  store.dispatch(settingActions.update('test'));
  expect(store.getActions()).toMatchSnapshot();
});

test('[action/soundActions]', () => {
  const store = configureStore([])({});
  store.dispatch(soundActions.request());
  store.dispatch(soundActions.requestSuccess('test'));
  store.dispatch(soundActions.mute());
  store.dispatch(soundActions.play('test'));
  store.dispatch(soundActions.throttleVolume('test', 'test'));
  store.dispatch(soundActions.volume('test', 'test'));
  store.dispatch(soundActions.edit('test', 'test'));
  store.dispatch(soundActions.remove('test'));
  store.dispatch(soundActions.addLocal('test'));
  store.dispatch(soundActions.addSound('test', 'test'));
  store.dispatch(soundActions.addSoundDownloading('test'));
  store.dispatch(soundActions.addSoundComplete('test'));
  store.dispatch(soundActions.reset('test'));
  store.dispatch(soundActions.playlist('test'));
  store.dispatch(soundActions.createPlaylist());
  expect(store.getActions()).toMatchSnapshot();
});

test('[action/themeActions]', () => {
  const store = configureStore([])({});
  store.dispatch(themeActions.change('test', 'test'));
  expect(store.getActions()).toMatchSnapshot();
});

test('[action/notifyActions]', () => {
  const store = configureStore([])({});
  store.dispatch(notifyActions.send('test', 'test'));
  store.dispatch(notifyActions.notify('test', 'test'));
  store.dispatch(notifyActions.clear('test'));
  expect(store.getActions()).toMatchSnapshot();
});
