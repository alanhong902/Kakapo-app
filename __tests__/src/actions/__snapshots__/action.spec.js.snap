// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[action/notifyActions] 1`] = `
Array [
  Object {
    "payload": Object {
      "msg": "test",
      "time": "test",
    },
    "type": "NOTIFICATIONS_SEND",
  },
  Object {
    "payload": Object {
      "id": "test",
      "msg": "test",
    },
    "type": "NOTIFICATIONS_NOTIFY",
  },
  Object {
    "payload": Object {
      "id": "test",
    },
    "type": "NOTIFICATIONS_CLEAR",
  },
]
`;

exports[`[action/searchActions] 1`] = `
Array [
  Object {
    "payload": Object {
      "input": "test",
    },
    "type": "SEARCH_INPUT",
  },
  Object {
    "type": "SEARCH_KAKAPO",
  },
  Object {
    "payload": Object {
      "term": "test",
    },
    "type": "SEARCH_YOUTUBE",
  },
  Object {
    "payload": Object {
      "term": "test",
    },
    "type": "SEARCH_SOUNDCLOUD",
  },
  Object {
    "type": "SEARCH_REQUEST",
  },
  Object {
    "payload": Object {
      "items": "test",
      "service": "test",
    },
    "type": "SEARCH_REQUEST_SUCCESS",
  },
  Object {
    "payload": Object {
      "err": "test",
    },
    "type": "SEARCH_REQUEST_ERROR",
  },
]
`;

exports[`[action/settingActions] 1`] = `
Array [
  Object {
    "payload": Object {
      "bool": "test",
    },
    "type": "SETTINGS_MUTE",
  },
  Object {
    "payload": Object {
      "bool": "test",
    },
    "type": "SETTINGS_DOCK",
  },
  Object {
    "payload": Object {
      "bool": "test",
    },
    "type": "SETTINGS_DEVTOOLS",
  },
  Object {
    "payload": Object {
      "locale": "test",
    },
    "type": "SETTINGS_LANGUAGE",
  },
  Object {
    "payload": Object {
      "status": "test",
    },
    "type": "SETTINGS_UPDATE",
  },
]
`;

exports[`[action/soundActions] 1`] = `
Array [
  Object {
    "type": "SOUNDS_REQUEST",
  },
  Object {
    "payload": Object {
      "resp": "test",
    },
    "type": "SOUNDS_REQUEST_SUCCESS",
  },
  Object {
    "type": "SOUNDS_MUTE",
  },
  Object {
    "payload": Object {
      "sound": "test",
    },
    "type": "SOUNDS_PLAY",
  },
  Object {
    "payload": Object {
      "sound": "test",
      "volume": "test",
    },
    "type": "SOUNDS_THROTTLE_VOLUME",
  },
  Object {
    "payload": Object {
      "sound": "test",
      "volume": "test",
    },
    "type": "SOUNDS_VOLUME",
  },
  Object {
    "payload": Object {
      "data": "test",
      "sound": "test",
    },
    "type": "SOUNDS_EDIT",
  },
  Object {
    "payload": Object {
      "sound": "test",
    },
    "type": "SOUNDS_REMOVE",
  },
  Object {
    "payload": Object {
      "file": "test",
    },
    "type": "SOUNDS_ADD_LOCAL",
  },
  Object {
    "payload": Object {
      "data": "test",
      "service": "test",
    },
    "type": "SOUNDS_ADD_SOUND",
  },
  Object {
    "payload": Object {
      "sound": "test",
    },
    "type": "SOUNDS_ADD_SOUND_DOWNLOADING",
  },
  Object {
    "payload": Object {
      "sound": "test",
    },
    "type": "SOUNDS_ADD_SOUND_COMPLETE",
  },
  Object {
    "payload": Object {
      "clear": "test",
    },
    "type": "SOUNDS_RESET",
  },
  Object {
    "payload": Object {
      "id": "test",
    },
    "type": "SOUNDS_PLAYLIST",
  },
  Object {
    "type": "SOUNDS_CREATE_PLAYLIST",
  },
]
`;

exports[`[action/themeActions] 1`] = `
Array [
  Object {
    "payload": Object {
      "slotNo": "test",
      "swatch": "test",
    },
    "type": "THEMES_CHANGE",
  },
]
`;
